workflow/scripts/add_extra_components.py

```diff
@@ -119,6 +130,103 @@ def attach_storageunits(n, costs, elec_opts, investment_year):
         )
 
 
+def attach_phs_storageunits(n: pypsa.Network, elec_opts):
+    carriers = elec_opts["extendable_carriers"]["StorageUnit"]
+    carriers = [k for k in carriers if "PHS" in k]
+
+    for carrier in carriers:
+        max_hours = int(carrier.split("hr_")[0])
+
+        psh_resources = (
+            gpd.read_file(
+                f"./repo_data/psh/40-100-dam-height-{max_hours}hr-no-croplands-no-ephemeral-no-highways.gpkg"
```

Comment:
maybe change this to a snakemake file input, to be consistent with the rest of the workflow (and make it easier to update in case the file name changes)?